### PR TITLE
fix: added 404 for error in getRule api

### PR DESCRIPTION
**Pull request by ankitnayan**
fixes https://github.com/SigNoz/signoz/issues/1307

### CODE REVIEW

File: pkg/query-service/app/clickhouseReader/reader.go [hunk 1]
```diff
@@ -592,19 +592,23 @@ func (r *ClickHouseReader) GetRulesFromDB() (*[]model.RuleResponseItem, *model.A
...
 
-	err := r.localDB.Get(rule, query)
+	err = r.localDB.Get(rule, query)
 
 	zap.S().Info(query)
 
 	if err != nil {
 		zap.S().Debug("Error in processing sql query: ", err)
-		return nil, &model.ApiError{Typ: model.ErrorInternal, Err: err}
+		return nil, &model.ApiError{Typ: model.ErrorNotFound, Err: err}
```
Comment (srikanthccv) [inline pkg/query-service/app/clickhouseReader/reader.go:611]: Does non-nil err always mean no entry for rule?

**Review by srikanthccv — DISMISSED** [review]

File: pkg/query-service/app/clickhouseReader/reader.go (hunk 1 above)
Comment (ankitnayan) [reply pkg/query-service/app/clickhouseReader/reader.go:611]: The lib returns an `error` object which needs to be parsed for text to generate different error codes. But then I looked into the API, it does not have much scope for error except for other internal errors. We could return `404` by matching the exact string and `500` for the rest of the errors.

Comment (ankitnayan) [reply pkg/query-service/app/clickhouseReader/reader.go:611]: Should I update for the above?

Comment (srikanthccv) [reply pkg/query-service/app/clickhouseReader/reader.go:611]: I think it's just fine to return 404. I just wanted to know if there are any other common errors. 

> it does not have much scope for error except for other internal errors

You are right. I just checked this https://pkg.go.dev/github.com/jmoiron/sqlx@v1.3.4#DB.Get

**Review by srikanthccv — APPROVED** [review]

File: pkg/query-service/app/clickhouseReader/reader.go (hunk 1 above)
Comment (ankitnayan) [reply pkg/query-service/app/clickhouseReader/reader.go:626]: @srikanthccv I have changed the function used to get data and added multiple checks for different error types.

**Review by srikanthccv — APPROVED** [review]
Updated code is more clear. Thanks

**Pull request merged**